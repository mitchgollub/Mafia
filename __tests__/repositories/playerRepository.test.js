import PlayerRepository from '../../repositories/playerRepository';

jest.mock('../../lib/mongodb');
Math.random = jest.fn(() => 0.48);

describe('PlayerRepository', () => {
  test('Adds Player', async () => {
    const game = {
      code: 'MMMM',
      players: {
        current: [{ id: 1, role: 'Narrator', name: 'YOU', session: '' }],
        available: [
          { id: 2, role: 'Cop' },
          { id: 3, role: 'Mafia' },
          { id: 4, role: 'Mafia' },
          { id: 5, role: 'Doctor' },
          { id: 6, role: 'Town Watch' },
          { id: 7, role: 'Townsfolk' },
          { id: 8, role: 'Townsfolk' },
          { id: 9, role: 'Townsfolk' },
        ],
      },
    };

    const playerRequest = {
      code: 'MMMM',
      name: 'Mitch',
      session: '0000-0000',
    };

    const actual = await PlayerRepository.addPlayer(game, playerRequest);
    expect(actual).toMatchSnapshot();
  });

  test('Adds Player Returns Existing Player', async () => {
    const game = {
      code: 'MMMM',
      players: {
        current: [
          { id: 1, role: 'Narrator', name: 'YOU', session: '' },
          {
            description: 'Doctors have ...',
            id: 5,
            name: 'Mitch',
            role: 'Doctor',
            session: '0000-0000',
          },
        ],
        available: [
          { id: 2, role: 'Cop' },
          { id: 3, role: 'Mafia' },
          { id: 4, role: 'Mafia' },
          { id: 6, role: 'Town Watch' },
          { id: 7, role: 'Townsfolk' },
          { id: 8, role: 'Townsfolk' },
          { id: 9, role: 'Townsfolk' },
        ],
      },
    };

    const playerRequest = {
      code: 'MMMM',
      name: 'Mitch',
      session: '0000-0000',
    };

    const actual = await PlayerRepository.addPlayer(game, playerRequest);
    expect(actual).toMatchSnapshot();
  });

  test('Adds Player Returns Empty Player if none available', async () => {
    const game = {
      code: 'MMMM',
      players: {
        current: [{ id: 1, role: 'Narrator', name: 'YOU', session: '' }],
        available: [],
      },
    };

    const playerRequest = {
      code: 'MMMM',
      name: 'Mitch',
      session: '0000-0000',
    };

    const actual = await PlayerRepository.addPlayer(game, playerRequest);
    expect(actual).toMatchSnapshot();
  });
});
