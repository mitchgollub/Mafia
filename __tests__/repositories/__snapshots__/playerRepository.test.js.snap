// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerRepository Adds Player 1`] = `
Player {
  "description": "Doctors have the ability to save other players from a Mafia murder.  At Night Time, the narrator will direct you to select one player to save.  If you were successful, the narrator will declare that no one was murdered.",
  "id": 5,
  "name": "Mitch",
  "role": "Doctor",
  "session": "0000-0000",
}
`;

exports[`PlayerRepository Adds Player Returns Empty Player if none available 1`] = `
Player {
  "description": undefined,
  "id": -1,
  "name": "Mitch",
  "role": "Empty",
  "session": "0000-0000",
}
`;

exports[`PlayerRepository Adds Player Returns Existing Player 1`] = `
Object {
  "description": "Doctors have the ability to save other players from a Mafia murder.  At Night Time, the narrator will direct you to select one player to save.  If you were successful, the narrator will declare that no one was murdered.",
  "id": 5,
  "name": "Mitch",
  "role": "Doctor",
  "session": "0000-0000",
}
`;
